### PR TITLE
fix: remove dependency p-retry

### PR DESCRIPTION
**Pull request by ocavue**
Remove the dependency `p-retry` and replace it with a simple hand-written version. 

Reason to remove `p-retry`:

`p-retry` is an _ES Module_ package, using the new JS syntax like [`export default function`](https://unpkg.com/browse/p-retry@6.1.0/index.js). 

`p-retry` depends on another package called `retry`. `retry` is a _CommonJS_ package, using the old JS syntax like [`module.exports = require`](https://unpkg.com/browse/retry@0.13.1/index.js). 

In short, it's invalid for a _ES Module_ package to depend on a _CommonJS_ package, like what `p-retry` is doing. While some bundlers can handle such cases, it's a good idea to always use the valid way so that the package could work anywhere. 

I know this is confusing, but sadly this is what the JS ecosystem is today.

### CODE REVIEW

**Comment by FL33TW00D** [comment]
> Remove the dependency `p-retry` and replace it with a simple hand-written version.
> 
> Reason to remove `p-retry`:
> 
> `p-retry` is an _ES Module_ package, using the new JS syntax like [`export default function`](https://unpkg.com/browse/p-retry@6.1.0/index.js).
> 
> `p-retry` depends on another package called `retry`. `retry` is a _CommonJS_ package, using the old JS syntax like [`module.exports = require`](https://unpkg.com/browse/retry@0.13.1/index.js).
> 
> In short, it's invalid for a _ES Module_ package to depend on a _CommonJS_ package, like what `p-retry` is doing. While some bundlers can handle such cases, it's a good idea to always use the valid way so that the package could work anywhere.
> 
> I know this is confusing, but sadly this is what the JS ecosystem is today.

You're a wizard 🪄
